bow.go

```diff
@@ -84,112 +81,113 @@ type Bow interface {
-			err = errors.New("bow: empty Series name")
-			return
+			return nil, errors.New("bow: empty Series name")
 		}
 		if getTypeFromArrowType(s.Array.DataType()) == Unknown {
-			err = fmt.Errorf("bow: unsupported type: %s", s.Array.DataType().Name())
-			return
-		}
-		if int64(s.Array.Len()) != nrows {
-			err = fmt.Errorf("bow: Series '%s' has a length of %d, which is different from the previous ones",
-				s.Name, s.Array.Len())
-			return
+			return nil, fmt.Errorf(
+				"bow: unsupported type: %s",
+				s.Array.DataType().Name())
 		}
-		newField := arrow.Field{
-			Name: s.Name,
-			Type: s.Array.DataType(),
+		if int64(s.Array.Len()) != nRows {
+			return nil,
+				fmt.Errorf(
+					"bow: Series '%s' has a length of %d, which is different from the previous ones",
+					s.Name, s.Array.Len())
 		}
-		fields = append(fields, newField)
-		cols = append(cols, s.Array)
+		fields = append(fields, arrow.Field{Name: s.Name, Type: s.Array.DataType()})
+		arrays = append(arrays, s.Array)
 	}
+
 	schema := arrow.NewSchema(fields, nil)
-	bobow = &bow{
-		Record: array.NewRecord(schema, cols, nrows),
-	}
-	return
+
+	return &bow{Record: array.NewRecord(schema, arrays, nRows)}, nil
 }
 
-func NewBowFromColumnBasedInterfaces(columnsNames []string, types []Type, columns [][]interface{}) (bobow Bow, err error) {
-	if len(columnsNames) != len(columns) {
-		err = errors.New("bow: columnsNames and columns array lengths don't match")
-		return
+func NewBowFromColumnBasedInterfaces(colNames []string, colTypes []Type, colData [][]interface{}) (Bow, error) {
+	if len(colNames) != len(colData) {
+		return nil, errors.New("bow: colNames and colData array lengths don't match")
 	}
 
-	if types != nil && len(columnsNames) != len(types) {
-		err = errors.New("bow: columnsNames and types array lengths don't match")
-		return
+	if colTypes != nil && len(colNames) != len(colTypes) {
+		return nil, errors.New("bow: colNames and colTypes array lengths don't match")
 	}
 
-	series := make([]Series, len(columnsNames))
-	for i, name := range columnsNames {
-		if types != nil {
-			series[i], err = NewSeriesFromInterfaces(name, types[i], columns[i])
+	var err error
+	series := make([]Series, len(colNames))
+	for i, name := range colNames {
+		if colTypes != nil {
+			series[i], err = NewSeriesFromInterfaces(name, colTypes[i], colData[i])
 		} else {
-			series[i], err = NewSeriesFromInterfaces(name, Unknown, columns[i])
+			series[i], err = NewSeriesFromInterfaces(name, Unknown, colData[i])
 		}
 		if err != nil {
-			return
+			return nil, err
 		}
 	}
 	return NewBow(series...)
 }
 
-func NewBowFromRowBasedInterfaces(columnsNames []string, types []Type, rows [][]interface{}) (bobow Bow, err error) {
-	columnBasedRows := make([][]interface{}, len(columnsNames))
-	for column := range columnsNames {
-		columnBasedRows[column] = make([]interface{}, len(rows))
+func NewBowFromRowBasedInterfaces(colNames []string, colTypes []Type, rowData [][]interface{}) (Bow, error) {
```

Comment:
could you create a TODO for perf improvement here in Readme ? -> we use col based at the end but it's a dummy implem.